src/verifiers/documentStoreStatus/openAttestationEthereumDocumentStoreStatus.ts

```diff
@@ -0,0 +1,187 @@
+        documentStores.map(async (documentStore) => {
+          try {
+            const contract = createDocumentStoreContract(documentStore, options);
+            const issued = await isIssuedOnDocumentStore(contract, merkleRoot);
+            const status: IssuanceStatus = {
+              issued,
+              address: documentStore,
+            };
+            if (!issued) {
+              status.reason = contractNotIssued(merkleRoot, documentStore);
+            }
+            return status;
+          } catch (e) {
+            return { issued: false, address: documentStore, reason: getErrorReason(e, documentStore) };
+          }
+        })
+      );
+      const notIssued = issuanceStatuses.find((status) => !status.issued);
+      if (notIssued) {
+        return {
+          name,
+          type,
+          data: {
+            issuedOnAll: false,
+            details: utils.isWrappedV3Document(document)
+              ? { issuance: issuanceStatuses[0] }
+              : { issuance: issuanceStatuses },
+          },
+          reason: notIssued.reason,
+          status: "INVALID",
+        };
+      }
+
+      const revocationStatuses: RevocationStatus[] = await Promise.all(
+        documentStores.map(async (documentStore) => {
+          try {
+            const contract = createDocumentStoreContract(documentStore, options);
+            const intermediateHashes = getIntermediateHashes(targetHash, proofs);
+            const revokedHash = await isAnyHashRevoked(contract, intermediateHashes);
+
+            const status: RevocationStatus = {
+              revoked: !!revokedHash,
+              address: documentStore,
+            };
+            if (revokedHash) {
+              status.reason = contractRevoked(merkleRoot, documentStore);
+            }
+            return status;
+          } catch (e) {
+            return { revoked: true, address: documentStore, reason: getErrorReason(e, documentStore) };
+          }
+        })
+      );
+      const revoked = revocationStatuses.find((status) => status.revoked);
+      if (revoked) {
+        return {
+          name,
+          type,
+          data: {
+            issuedOnAll: true,
+            revokedOnAny: true,
+            details: utils.isWrappedV3Document(document)
+              ? { issuance: issuanceStatuses[0], revocation: revocationStatuses[0] }
+              : { issuance: issuanceStatuses, revocation: revocationStatuses },
+          },
+          reason: revoked.reason,
+          status: "INVALID",
+        };
+      }
+
+      return {
+        name,
+        type,
+        data: {
+          issuedOnAll: true,
+          revokedOnAny: false,
+          details: utils.isWrappedV3Document(document)
+            ? { issuance: issuanceStatuses[0], revocation: revocationStatuses[0] }
+            : { issuance: issuanceStatuses, revocation: revocationStatuses },
+        },
+        status: "VALID",
+      };
```

Comment:
```suggestion
      return {
        name,
        type,
        data: {
          issuedOnAll: true,
          revokedOnAny: !!revoked,
          details: utils.isWrappedV3Document(document)
            ? { issuance: issuanceStatuses[0], revocation: revocationStatuses[0] }
            : { issuance: issuanceStatuses, revocation: revocationStatuses },
        },
        ...(revoked ? { reason: revoked.reason } : {}),
        status: revoked ? "INVALID" : "VALID",
      };
```

Suggestion, feel free to ignore :)